packages/frontend/src/app/components/checkout/CartButton.tsx

```diff
@@ -2,16 +2,27 @@
 //
 // SPDX-License-Identifier: GPL-3.0-or-later
 
-import React from "react";
+import React, { useState, useEffect } from "react";
 import Image from "next/image";
 import { useStoreContext } from "@/context/StoreContext";
 import Link from "next/link";
+import { ItemState } from "@/context/types";
 
 const CartButton = () => {
-  const { orderItems, orderId } = useStoreContext();
+  const { stateManager, getOrderId } = useStoreContext();
+  const [cartItemIds, setItemIds] = useState<ItemState | null>(null);
 
-  const activeCartItems = orderId && orderItems.get(orderId)?.items;
-  const arr = activeCartItems ? Object.values(activeCartItems) : [];
+  useEffect(() => {
+    (async () => {
+      if (stateManager) {
```

Comment:
we could instantiate the stateManager with an initial loading value like LoadingStateManager instead of with null. This way we wont have to check whether stateManager is truthy everytime. We would just have to return a new promise with initial values for each state, manifest, products, orders, tags, etc.


```
class LoadingManifestManager<T extends ShopObjectTypes> extends EventEmitter {
  constructor() {
    super();
  }
  async get(key?: string): Promise<T> {
    return new Promise(() => {
      return { name: "", publishedTagId: null };
    });
  }
  async iterator() {
    return;
  }
}
export class LoadingStateManager {
  readonly items;
  readonly tags;
  readonly manifest;
  readonly orders;
  readonly keycards;
  constructor() {
    this.items = new LoadingManager();
    this.tags = new LoadingManager();
    this.manifest = new LoadingManifestManager();
    this.orders = new LoadingManager();
    this.keycards = new LoadingManager();
  }
}
```

Reply:
@cryptix okay I implemented the initial loading state for statemanager. take a look at commit [0e5b4a0]

Reply:
@cryptix @jchoi330 I'm not sure what the best way to do this is either